### PR TITLE
New benchmark configurations

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

Adjust our current benchmarks configurations, in order to reduce the costs and make the benchmarks more meaningful. 

The benchmarks are now aligned to G3-S cluster plan, which is also used in SaaS. This makes them more realistic. Furthermore the default load was adjusted so we run a real throughput tests, to verify what is the maximum we can achieve with this resources.

The mixed benchmark workflow have been adjusted to setup, a default throughput test, a mixed benchmark (similar load to default) but using more symbols and a latency tests. The latency tests runs one process instance per second and allows to determine how our process instance execution time is over time.

### New throughput test:

In our recent throughput test we can see with the G3-S adjustment we are able to execute and complete ~60-65 instances per second.

throughput-overview

**Processing queue**
throughput-processing

**Completions per second**
throughput-completions

### New Latency test

In our newest latency test we can see that we can execute a process instance in around 100-200 ms, but there are outliers which go up to 1.5 sec. This is something we should investigate.

latency-overview

**Latency**
latency-latency

**Latency last 6h**
latency-latency-6h

latency-latency-commit-6h

## Related issues

closes #

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: .github/workflows/medic-benchmarks.yml [hunk 1]
```diff
@@ -54,5 +54,17 @@ jobs:
...
+      - uses: ./.github/workflows/benchmark.yaml
+        name: Create latency benchmark
+        with:
+          name: medic-cw-${{ steps.data.outputs.cw }}-${{ steps.data.outputs.ref }}-benchmark-latency
+          cluster: zeebe-cluster
+          cluster-region: europe-west1-b
+          ref: ${{ steps.data.outputs.ref }}
+          load: "worker, starter=1"
+        secrets: inherit
+      - name: Downscale workers on latency benchmark
```
Comment (Zelldon) [inline .github/workflows/medic-benchmarks.yml:68]: :question: Here I was unsure whether the inherited resources/setup are here still available? Maybe you know that @oleschoenburg

Comment (oleschoenburg) [reply .github/workflows/medic-benchmarks.yml:68]: Here we don't have a working `kubectl` so the downscaling won't work. You could either:
* copy some of the setup steps from the benchmark workflow that setup gcloud and authenticate with the cluster
* support `worker = 1` syntax in the benchmark workflow to control the number of replicas.
* support arbitrary k8s json patches in the benchmark workflow. Here we could use it to patch the workers replica count.

Comment (Zelldon) [reply .github/workflows/medic-benchmarks.yml]: We migrated to https://github.com/camunda/zeebe/pull/11155 helm benchmarks :rocket:

**Comment by Zelldon** [comment]
I updated this PR to our recent changes @oleschoenburg please have a look :bow:

**Review by lenaschoenburg — APPROVED** [review]
Looks good to me overall :+1: 

:thought_balloon: One difference between this (and the [values](https://github.com/zeebe-io/benchmark-helm/blob/main/zeebe-benchmark/values.yaml) file) and the G3-S plan is the disk size. Should we adjust that too? I think it affects max IOPS which we know is a bottleneck for Zeebe.

**Comment by Zelldon** [comment]
I was also thinking about that. In saas we have the autoscaling, Idk how to do that in our benchmarks right now. The limit is 192GB and starts with 32G, so since it is somehow in between I think it is ok for now.

**Pull request merged**